process/interface.go

```diff
@@ -151,6 +151,13 @@ type BlockProcessor interface {
 	IsInterfaceNil() bool
 }
 
+// PeerProcessor is the main interface for validators' consensus participation statistics
+type PeerProcessor interface {
+	LoadInitialState(in []*sharding.InitialNode) error
+	UpdatePeerState(header, previousHeader data.HeaderHandler) error
```

Comment:
Are these functions enough ? what happens in case of rollback ?

Reply:
You are right, in the following pull request on the shard I also updated the metachain to be able to rollback. Actually I will add a function that takes a header, and does the opposite from this update function (decrease if validator signed, increase if he did not) - if it seems ok for you too